pmd-scala/src/main/java/net/sourceforge/pmd/lang/scala/ast/ScalaWrapperNode.java

```diff
@@ -0,0 +1,155 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.scala.ast;
+
+import java.util.Objects;
+
+import net.sourceforge.pmd.lang.ast.AbstractNode;
+import net.sourceforge.pmd.lang.ast.Node;
+import net.sourceforge.pmd.lang.scala.ScalaParser;
+
+import scala.meta.Tree;
+import scala.meta.inputs.Position;
+
+/**
+ * The Java-wrapper for Scala Nodes to allow interoperability with PMD.
+ */
+public class ScalaWrapperNode extends AbstractNode implements ScalaNode {
+    private Tree node;
+    private ScalaParser parser;
+
+    /**
+     * Create a new Java Wrapper around a Scala node.
+     * 
+     * @param scalaParser
+     *            the ScalaParser used to generate the node
+     * @param scalaNode
+     *            the scalaNode node to wrap
+     */
+    public ScalaWrapperNode(ScalaParser scalaParser, Tree scalaNode) {
+        super(0);
+        this.parser = scalaParser;
+        this.node = scalaNode;
+        Position pos = node.pos();
+        beginLine = pos.startLine() + 1;
+        endLine = pos.endLine() + 1;
+        beginColumn = pos.startColumn() + 1;
+        endColumn = pos.endColumn() + 1;
+    }
+
+    @Override
+    public Tree getNode() {
+        return node;
+    }
+
+    @Override
+    public Object accept(ScalaParserVisitor visitor, Object data) {
+        return visitor.visit(this, data);
+    }
+
+    @Override
+    public Object childrenAccept(ScalaParserVisitor visitor, Object data) {
+        int numChildren = jjtGetNumChildren();
+        for (int i = 0; i < numChildren; ++i) {
+            jjtGetChild(i).accept(visitor, data);
+        }
+        return data;
+    }
+
+    @Override
+    public String getXPathNodeName() {
+        return "AST" + node.productPrefix().replace(".", "") + "Node";
```

Comment:
We maybe don't need the "Node" suffix, since it's always a node. Same with "AST" prefix.
This is a API decision, and whatever we decide to go with, we need to live for a long time.
I know, that we have for the other languages "AST" prefix, but that was probably because we use the (Java) class name and prefixed the classes with "AST" in order to find them easier.

In the end, we should choose something, that we think is easier to read later on - "AST" and "Node" would only be superfluous and distract only?

Reply:
If this is acceptable to you, I'd much prefer not to add extraneous info to the tree nodes. I was following the guidance from the docs: https://pmd.github.io/latest/pmd_devdocs_major_adding_new_language.html but I agree that both AST and Node are unnecessary additions. I'll remove this.

Reply:
This section actually talks about the class names in Java, which should use the AST prefix. But here we are deciding on the names of the nodes in the AST, that XPath uses... at some places, we simply use the class name and remove the AST prefix (e.g. in the Apex language). So, not having AST here, is consistent.